### PR TITLE
feat: add fork component on workflow

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5922.0 ✘  
* Stage 1
  * Engine Test with DB ✘

File: engine/api/workflow/dao_fork.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	forks := make([]sdk.WorkflowNodeFork, len(res))
+	for i := range res {
+		res[i].WorkflowNodeID = node.ID
+		forks[i] = sdk.WorkflowNodeFork(res[i])
+
+		//Select triggers id
+		var triggerIDs []int64
+		if _, err := db.Select(&triggerIDs, "select id from workflow_node_fork_trigger where workflow_node_fork_id = $1", forks[i].ID); err != nil {
+			if err == sql.ErrNoRows {
+				return nil, sdk.WrapError(err, "loadForks> Unable to load fork triggers id for for %d", forks[i].ID)
```
Comment (bnjjj) [inline engine/api/workflow/dao_fork.go]: I think there is an error here

File: ui/src/app/model/workflow.model.ts [hunk 2]
```diff
@@ -516,9 +608,98 @@ export class WorkflowNode {
...
+    static removeForkWithoutChild(node: WorkflowNode, id: number): boolean {
+        if (node.forks && node.forks.length > 0) {
+            for (let i = 0; i < node.forks.length; i ++) {
+                if (node.forks[i].id === id) {
+                    // create node trigger for each nodefork trigger
+                    if (node.forks[i].triggers) {
+                        if (!node.triggers) {
+                            node.triggers = new Array<WorkflowNodeTrigger>();
+                        }
+                        node.forks[i].triggers.forEach(t => {
```
Comment (bnjjj) [inline ui/src/app/model/workflow.model.ts]: use map

File: ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                        <button class="ui blue button"><i class="close icon"
+                                                          (click)="displayInputName = false; fork.name = previousNodeName"></i>
+                        </button>
+                    </div>
+                </div>
+            </ng-container>
+        </div>
+    </h3>
+    <div class="ui fluid vertical menu noborder">
+        <!--a class="item noborder"
```
Comment (bnjjj) [inline ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.html]: to delete ?

File: ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        }
+        this.displayInputName = true;
+    }
+
+    saveTrigger(): void {
+        if (this.workflow.permission !== PermissionValue.READ_WRITE_EXECUTE) {
+            return null
+        }
+        let clonedWorkflow: Workflow = cloneDeep(this.workflow);
+        let currentFork: WorkflowNodeFork;
```
Comment (bnjjj) [inline ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts]: `let currentFork = Workflow.getForkByID(this.fork.id, clonedWorkflow);`

File: ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        t.workflow_dest_node = this.newTrigger;
+        if (!currentFork.triggers) {
+            currentFork.triggers = new Array<WorkflowNodeForkTrigger>();
+        }
+        currentFork.triggers.push(t);
+        this.updateWorkflow(clonedWorkflow, this.workflowTrigger.modal, true);
+    }
+
+    rename(): void {
+
```
Comment (bnjjj) [inline ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts]: delete

File: ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        if (this.workflow.permission !== PermissionValue.READ_WRITE_EXECUTE) {
+            return;
+        }
+        let clonedWorkflow: Workflow = cloneDeep(this.workflow);
+        this.updateWorkflow(clonedWorkflow, null, false);
+
+
```
Comment (bnjjj) [inline ui/src/app/views/workflow/sidebar/edit/fork/workflow.sidebar.fork.component.ts]: delete

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5923.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5925.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5931.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Pull request merged**